python/dgl/data/csv_dataset.py

```diff
@@ -0,0 +1,427 @@
+import os
+import yaml
+from yaml.loader import SafeLoader
+import pandas as pd
+import numpy as np
+from typing import List, Optional
+import pydantic as dt
+from .dgl_dataset import DGLDataset
+from ..convert import heterograph as dgl_heterograph
+from .. import backend as F
+from .utils import save_graphs, load_graphs
+from ..base import dgl_warning, DGLError
+import abc
+import ast
+from typing import Callable
+
+
+class MetaNode(dt.BaseModel):
+    """ Class of node_data in YAML. Internal use only. """
+    file_name: str
+    ntype: Optional[str] = '_V'
+    graph_id_field: Optional[str] = 'graph_id'
+    node_id_field: Optional[str] = 'node_id'
+
+
+class MetaEdge(dt.BaseModel):
+    """ Class of edge_data in YAML. Internal use only. """
+    file_name: str
+    etype: Optional[List[str]] = ['_V', '_E', '_V']
+    graph_id_field: Optional[str] = 'graph_id'
+    src_id_field: Optional[str] = 'src_id'
+    dst_id_field: Optional[str] = 'dst_id'
+
+
+class MetaGraph(dt.BaseModel):
+    """ Class of graph_data in YAML. Internal use only. """
+    file_name: str
+    graph_id_field: Optional[str] = 'graph_id'
+
+
+class MetaYaml(dt.BaseModel):
+    """ Class of YAML. Internal use only. """
+    version: Optional[str] = '1.0.0'
+    dataset_name: str
+    separator: Optional[str] = ','
+    node_data: List[MetaNode]
+    edge_data: List[MetaEdge]
+    graph_data: Optional[MetaGraph] = None
+
+
+def load_yaml_with_sanity_check(yaml_file):
+    """ Load yaml and do sanity check. Internal use only. """
+    with open(yaml_file) as f:
+        yaml_data = yaml.load(f, Loader=SafeLoader)
+        try:
+            meta_yaml = MetaYaml(**yaml_data)
+        except dt.ValidationError as e:
+            print(
+                "Details of pydantic.ValidationError:\n{}".format(e.json()))
+            raise DGLError(
+                "Validation Error for YAML fields. Details are shown above.")
+        if meta_yaml.version != '1.0.0':
+            raise DGLError("Invalid CSVDataset version {}. Supported versions: '1.0.0'".format(
+                meta_yaml.version))
+        ntypes = [meta.ntype for meta in meta_yaml.node_data]
+        if len(ntypes) > len(set(ntypes)):
+            raise DGLError(
+                "Duplicate node types for different CSV files which is required to have unique node type. ntypes: {}.".format(ntypes))
```

Comment:
`'Each node CSV file must have a unique node type name, but found duplicate node type "{}".'

Reply:
print the duplicate ones only? or all node types?